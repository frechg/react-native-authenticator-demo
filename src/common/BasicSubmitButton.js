import * as React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import { useFormikContext } from 'formik';
import { styles } from './styles';

export const BasicSubmitButton = ({title, ...props}) => {
  /*
  const {
    formik: { handleSubmit, isSubmitting },
    ...buttonProps
  } = props
  */

  const { handleSubmit, isSubmitting } = useFormikContext();

  return (
    <TouchableOpacity style={styles.buttonPrimary} title={title} onPress={handleSubmit} {...props}>
      <Text style={styles.buttonPrimaryText}>
        { isSubmitting ? 'Submiting...' : title }
      </Text>
    </TouchableOpacity>
  )
}