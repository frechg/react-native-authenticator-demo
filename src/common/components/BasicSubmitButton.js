import * as React from 'react';
import { TouchableOpacity, Text, ActivityIndicator } from 'react-native';
import { useFormikContext } from 'formik';
import { styles } from '../styles';

export const BasicSubmitButton = ({title, ...props}) => {
  const { handleSubmit, isSubmitting } = useFormikContext();

  return (
    <TouchableOpacity style={styles.buttonPrimary} title={title} onPress={handleSubmit} {...props}>
      { isSubmitting ? (
          <ActivityIndicator size='small' style={{marginVertical: 1}}/>
        ) :
          <Text style={styles.buttonPrimaryText}>{title}</Text>
      }
    </TouchableOpacity>
  )
}