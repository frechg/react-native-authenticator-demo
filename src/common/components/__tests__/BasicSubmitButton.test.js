import * as React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react-native';
import { useFormikContext } from 'formik';
import { BasicSubmitButton } from '../BasicSubmitButton';

jest.mock('formik');

describe('<BasicSubmitButton />', () => {
  test('Renders a basic submit button', () => {
    const handleSubmit = jest.fn();
    const isSubmitting = false;
    useFormikContext.mockReturnValue({handleSubmit, isSubmitting});

    const title = 'Test Submit';
    render(
      <BasicSubmitButton title={title} />
    )

    expect(screen.getByText(title)).toBeTruthy();
    expect(screen).toMatchSnapshot();
  });

  test('Calls the submit handler passed by the Formik context', async () => {
    const handleSubmit = jest.fn();
    const isSubmitting = false;
    useFormikContext.mockReturnValue({handleSubmit, isSubmitting});

    render(
      <BasicSubmitButton title='Submit Test' />
    )
    
    fireEvent.press(screen.getByText('Submit Test'));

    await waitFor(() => {
      expect(handleSubmit).toHaveBeenCalled();
    });
  });

  test('Hides button title and shows loading indicator when submitting', async () => {
    const handleSubmit = jest.fn();
    const isSubmitting = true;
    useFormikContext.mockReturnValue({handleSubmit, isSubmitting});

    render(
      <BasicSubmitButton title='Submit Test' />
    )
    
    await waitFor(() => {
      expect(screen.getByText('Submitting...')).toBeTruthy();
    });
  });
});