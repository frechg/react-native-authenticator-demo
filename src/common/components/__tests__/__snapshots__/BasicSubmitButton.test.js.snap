// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicSubmitButton /> Hides button title and shows loading indicator when submitting 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(40,40,40)",
      "borderRadius": 6,
      "opacity": 1,
      "padding": 12,
    }
  }
>
  <ActivityIndicator
    size="small"
    style={
      Object {
        "marginVertical": 1,
      }
    }
  />
</View>
`;

exports[`<BasicSubmitButton /> Renders a basic submit button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(40,40,40)",
      "borderRadius": 6,
      "opacity": 1,
      "padding": 12,
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(255,255,255)",
        "fontSize": 18,
      }
    }
  >
    Test Submit
  </Text>
</View>
`;
